src/almanack/batch.py

```diff
@@ -0,0 +1,235 @@
+"""
```

Comment:
I recommend calling this file `batch_processing.py` - otherwise, it is unclear by the name of the file what a "batch" is.